calcom/cal.com
# feat: autosubmit at all digits to enable 2fa

## What does this PR do?

Adds Auto-submit to enable 2FA.

Fixes #9726 

## Demo:
![success](https://github.com/calcom/cal.com/assets/78294692/734b333c-7c79-4a1d-bc56-d6606ecb0643)

## Type of change

- New feature
## Tested the following:

- enable 2FA, check payload and response [200]
- disable 2FA, check payload and response [200]

**tl;dr changes**: added `optional?` params to `TwoFactor.tsx` to handle submit, using just **useEffect** to check for the length of 2fa code with change in the overall `value` of the code to prevent multiple post requests.

### Notes for maintainer:
- Spotted a bug in disabling 2FA, as you can see towards the end of the video above, 2 "Two-Factor Code" text appears:
![image](https://github.com/calcom/cal.com/assets/78294692/6b220317-5343-43ad-9d54-b284274a0ed7)
Created an issue for it: #9761 

/claim #9726
